sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFs.scala

```diff
@@ -235,6 +200,56 @@ private[hive] case class HiveGenericUDF(
   }
 }
 
+class HiveGenericUDFHelper(
+    funcWrapper: HiveFunctionWrapper, children: Seq[Expression])
+  extends HiveInspectors
+  with Serializable
+  with Logging {
+
+  @transient
+  private[hive] val deterministic = isUDFDeterministic && children.forall(_.deterministic)
+
+  @transient
+  private[hive] val foldable =
```

Comment:
why do we define these properties here instead of in the expression? are we going to reuse it in `HiveSimpleUDF`?

Reply:
If we do not define `dataType` `foldable` `deterministic` properties in this class, we will have to expose some internal logic, such as `returnInspector`
<img width="723" alt="image" src="https://user-images.githubusercontent.com/15246973/224871989-7be999b4-c229-44d1-9e53-658ecacc539c.png">

With the similar implementation of HiveSimpleUDF, I found some common properties, or I can do some abstraction and reuse later.

Reply:
SGTM

Reply:
When I am working on `HiveSimpleUDF `, I will try to do this.